hyperspy/misc/array_tools.py

```diff
@@ -163,6 +166,10 @@ def rebin(a, new_shape=None, scale=None, crop=True):
     # check whether or not interpolation is needed.
     if _requires_linear_rebin(arr=a, scale=scale):
         _logger.debug("Using linear_bin")
+        if np.issubdtype(a.dtype, np.integer):
+            # The _linear_bin function below requires a float dtype
+            # because of the default numpy casting rule ('same_kind').
```

Comment:
Where exactly does the casting take place? I am not sure if the default is actually `same_kind`.

Reply:
I'm not sure how well I understand the error having not looked at this in a while. However it was maintaining the dtype during linear binning which was causing the resin error that came up with weird results. A floating dtype is needed to preserve counts where we're binning in such a way to split up pixels.